components/event-sources/reconciler/httpsource/reconcile.go

```diff
@@ -283,12 +356,31 @@ func (r *Reconciler) syncChannel(src *sourcesv1alpha1.HTTPSource,
 	return ch, nil
 }
 
+// syncPolicy synchronizes the desired state of a Policy against its current
+// state in the running cluster.
+func (r *Reconciler) syncPolicy(src *sourcesv1alpha1.HTTPSource,
+	currentPolicy, desiredPolicy *authenticationv1alpha1.Policy) (*authenticationv1alpha1.Policy, error) {
+
+	if object.Semantic.DeepEqual(currentPolicy, desiredPolicy) {
+		return currentPolicy, nil
+	}
+
+	policy, err := r.authClient.Policies(currentPolicy.Namespace).Update(desiredPolicy)
+	if err != nil {
+		r.eventWarn(src, failedUpdateReason, "Update failed for Policy %q", desiredPolicy.Name)
+		return nil, err
+	}
+	r.event(src, updateReason, "Updated Policy %q", policy.Name)
```

Comment:
@sayanh @antoineco Do we need to log an event with every time the policy is updated?

Reply:
I believe it is useful to see when something was changed by the controller using `kubectl describe`, providing it is something that is expected to change rarely.